### PR TITLE
TST: User prompted if auth set, credential not set

### PR DESCRIPTION
**Pull request by debanjum**
refer:  issue #666, pr #665

### CODE REVIEW

**Comment by yarikoptic** [comment]
Oh. . We seems to need use some fake keyring... I will give detail later if you dummy fund it before used in other tests

File: datalad/downloaders/tests/test_http.py [hunk 1]
```diff
@@ -351,6 +351,13 @@ def request_get_callback(request, uri, headers):
     # the provided URL at the end 404s, or another failure (e.g. interrupted download)
 
 
+@with_testsui(responses=['no', 'yes', 'testlogin', 'testpassword'])
+def test_auth_but_no_cred():
+    authenticator = HTMLFormAuthenticator("")
+    assert_raises(ValueError, HTTPDownloader, credential=None, authenticator=authenticator)
```
Comment (yarikoptic) [inline datalad/downloaders/tests/test_http.py]: ok -- create `@with_memory_keyring` decorator, which would mock out `Credential._keyring` to be a `MemoryKeyring()`.  We use that one in `test_credentials` but I never made a decorator for such purpose.  Could be not a decorator but e.g. a context manager so you could do `with memory_keyring as keyring:` and then even interrogate that `keyring` regarding what was stored etc.

**Comment by yarikoptic** [comment]
Didn't we figure it all out @debanjum ? ;-)

**Comment by yarikoptic** [comment]
Note that skill test platforms fail here

**Comment by debanjum** [comment]
It seems like composite_credentials [failed](https://travis-ci.org/datalad/datalad/jobs/149555559#L2156) because it doesn't know new `MemoryKeyring` structure

**Comment by yarikoptic** [comment]
please merge master and push again -- build with freshier annex (on osx) must be fixed now

**Comment by yarikoptic** [comment]
On Wed, 03 Aug 2016, Debanjum wrote:

>    It seems like composite_credentials failed because it doesn't know new
>    MemoryKeyring structure

could you fix up for it, or should I have a look?

**Comment by debanjum** [comment]
I think I fixed the issue given that the travis build didn't fail the previous time

**Pull request merged**